### PR TITLE
chore: move kuma-net inside kuma repository

### PR DESCRIPTION
**Pull request by bartsmykla**
Move kuma-net inside kuma repo

Closes: https://github.com/kumahq/kuma-net/issues/113

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [ ] Manual Universal Tests --
- [ ] Manual Kubernetes Tests --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by mmorel-35** [comment]
Hi @bartsmykla ,

Have you considered removing 
https://github.com/bartsmykla/kuma/blob/chore/move-kuma-net-inside-kuma/.golangci.yml#L42

since github.com/kumahq/kuma-net won't exist anymore.

File: mk/build.mk [hunk 1]
```diff
@@ -28,8 +28,12 @@ COREDNS_VERSION ?= v1.10.0
 COREDNS_TMP_DIRECTORY ?= $(BUILD_DIR)/coredns
 COREDNS_PLUGIN_CFG_PATH ?= $(TOP)/tools/builds/coredns/templates/plugin.cfg
 
+EBPF_GIT_REPOSITORY ?= https://github.com/kumahq/merbridge.git
+EBPF_GIT_BRANCH ?= main
+EBPF_TMP_DIRECTORY ?= $(BUILD_DIR)/ebpf
+
 # List of binaries that we have release build rules for.
-BUILD_RELEASE_BINARIES := kuma-cp kuma-dp kumactl coredns envoy kuma-cni install-cni
+BUILD_RELEASE_BINARIES := ebpf/copy-for-kumactl kuma-cp kuma-dp kumactl coredns envoy kuma-cni install-cni
```
Comment (lahabana) [inline mk/build.mk:36]: This seems not required (you dep on it in build/kumactl already)

File: test/framework/kuma-net/socket/port.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+	for port := range GenerateRandomPorts(num, restrictedPort...) {
+		ports = append(ports, port)
+	}
+
+	return ports
+}
+
+func GenerateRandomPorts(num uint, restrictedPort ...uint16) map[uint16]struct{} {
+	rand.Seed(time.Now().UnixNano())
```
Comment (lahabana) [inline test/framework/kuma-net/socket/port.go:20]: Should use gomega or gingko's seed (makes tests reproducible)

File: mk/build.mk (hunk 1 above)
Comment (bartsmykla) [reply mk/build.mk]: addressed

**Comment by bartsmykla** [comment]
@mmorel-35 thanks for catching this - fixed

File: tools/releases/dockerfiles/Dockerfile.kuma-init [hunk 3]
```diff
@@ -17,9 +15,6 @@ COPY /tools/releases/templates/LICENSE \
 
 COPY /tools/releases/templates/NOTICE-kumactl /kuma/NOTICE
 
-# ebpf
-COPY --from=ebpf /ebpf /kuma/ebpf
```
Comment (lukidzi) [inline tools/releases/dockerfiles/Dockerfile.kuma-init:21]: Only have one question. How do we get ebpf files copied to the image now?

Comment (bartsmykla) [reply tools/releases/dockerfiles/Dockerfile.kuma-init:21]: they are compiled into `kumactl` binary and then when installing tproxy, they are taken from the binary and saved to the filesystem

File: test/blackbox_tests/blackbox_tests_suite_test.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestTransparentProxy(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Blackbox tests Suite")
```
Comment (michaelbeaumont) [inline test/blackbox_tests/blackbox_tests_suite_test.go:12]: I think "black box tests" everywhere should be something like "kuma-net tests" or "ebpf tests"

File: mk/build.mk [hunk 5]
```diff
@@ -105,6 +109,60 @@ else
 	@echo "CoreDNS is already built. If you want to rebuild it, remove the binary: rm $(BUILD_ARTIFACTS_DIR)/coredns/coredns"
 endif
 
+foo:
```
Comment (michaelbeaumont) [inline mk/build.mk:112]: should this be here?

File: mk/build.mk [hunk 6]
```diff
@@ -28,6 +28,10 @@ COREDNS_VERSION ?= v1.10.0
 COREDNS_TMP_DIRECTORY ?= $(BUILD_DIR)/coredns
 COREDNS_PLUGIN_CFG_PATH ?= $(TOP)/tools/builds/coredns/templates/plugin.cfg
 
+EBPF_GIT_REPOSITORY ?= https://github.com/kumahq/merbridge.git
+EBPF_GIT_BRANCH ?= main
```
Comment (michaelbeaumont) [inline mk/build.mk:32]: shouldn't we be explicitly tracking a specific version?

File: test/framework/kuma-net/udp/server.go [hunk 7]
```diff
@@ -0,0 +1,120 @@
+package udp
```
Comment (michaelbeaumont) [inline test/framework/kuma-net/udp/server.go:1]: I don't think we want to use "kuma-net" anymore, right? so rename this directory, for example.

**Review by michaelbeaumont — COMMENTED** [review]
Is everything under `pkg/transparentproxy` and `test/framework/kuma-net` just copy pasted from kuma-net?

File: mk/build.mk (hunk 5 above)
Comment (bartsmykla) [reply mk/build.mk:112]: definitely not, thank you for catching this!

File: mk/build.mk (hunk 6 above)
Comment (bartsmykla) [reply mk/build.mk:32]: at this point we are not directly mirroring merbridge tree, and the plan is to upstream the changes, so I would leave it as is, and if anything would change, we can pin the specific version. Please open the conversation back if you disagree

File: test/blackbox_tests/blackbox_tests_suite_test.go (hunk 4 above)
Comment (bartsmykla) [reply test/blackbox_tests/blackbox_tests_suite_test.go]: I called them as I thought in the future we might want to write blackbox tests for some other (non network related) cases, but I agree, that we can change it in the future, so I renamed `blackbox_tests` to `blackbox_network_tests`. Wdyt about it?

File: mk/build.mk (hunk 5 above)
Comment (bartsmykla) [reply mk/build.mk]: addressed

File: test/framework/kuma-net/udp/server.go (hunk 7 above)
Comment (bartsmykla) [reply test/framework/kuma-net/udp/server.go]: renamed it to `test/framework/network`

**Comment by bartsmykla** [comment]
> Is everything under `pkg/transparentproxy` and `test/framework/kuma-net` just copy pasted from kuma-net?

The only difference in `pkg/transparentproxy` is in config directory

File: test/blackbox_tests/blackbox_tests_suite_test.go (hunk 4 above)
Comment (michaelbeaumont) [reply test/blackbox_tests/blackbox_tests_suite_test.go]: yeah that's clearer

**Review by lukidzi — APPROVED** [review]

**Pull request merged**